app/repositories/supabase_repository.py

```
@@ -72,6 +72,20 @@ def exists(self, device_id: str, business_id: str, biz_item_id: str) -> bool:
             return False
 
+    def count_by_device(self, device_id: str) -> int:
+        """
+        Retrieves the total number of favorite items for a device.
+        """
+        try:
+            response = self.supabase.table(self.table_name).select(
+                "*", count="exact", head=True
+            ).eq("device_id", device_id).execute()
+            
+            return response.count or 0
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)